packages/mongo-nav/src/MongoNav.spec.tsx

```diff
@@ -1,68 +1,310 @@
+    expectedElements.orgNav = queryByTestId('organization-nav');
+    expectedElements.projectNav = queryByTestId('project-nav');
+    expectedElements.admin = queryByTestId('org-nav-admin-link');
+    expectedElements.billing = queryByTestId('org-nav-billing');
+    expectedElements.activityFeed = queryByTestId('project-nav-activity-feed');
+    expectedElements.currentOrg = queryByTestId('org-select-active-org');
+    expectedElements.currentProject = queryByTestId(
+      'project-select-active-project',
+    );
+    expectedElements.userMenu = queryByTestId('user-menu-trigger');
+  };
+
+  let onOrganizationChange: jest.Mock;
+  let onProjectChange: jest.Mock;
+  let fetchMock: jest.Mock;
+
+  beforeEach(() => {
+    onOrganizationChange = jest.fn();
+    onProjectChange = jest.fn();
+    fetchMock = jest.fn();
+    window.fetch = fetchMock;
+  });
+
+  afterEach(() => {
+    delete window.fetch;
+    jest.restoreAllMocks();
+    cleanup();
+  });
+
+  const renderComponent = (props = {}) => {
+    setQueries(
+      render(
+        <MongoNav
+          activeProduct="cloud"
+          onOrganizationChange={onOrganizationChange}
+          onProjectChange={onProjectChange}
+          {...props}
+        />,
+      ),
+    );
+  };
 
   describe('by default', () => {
-    test('it renders the ProjectNav', () => {
-      const projectNav = getByTestId('project-nav');
-      expect(projectNav).toBeInTheDocument();
+    const responseObject = {
+      ok: true,
+      json: () => Promise.resolve(dataFixtures),
+    };
+
+    beforeEach(async () => {
+      fetchMock.mockResolvedValue(responseObject);
+      renderComponent();
+
+      await act(() => responseObject.json());
     });
 
-    test('it renders admin as false', () => {
-      const orgNav = getByTestId('organization-nav');
-      expect(orgNav.innerHTML.includes('Admin')).toBe(false);
+    test('fetch is called', () => {
+      expect(fetchMock).toHaveBeenCalledTimes(1);
+    });
+
+    test('the organization nav is rendered', () => {
+      const organizationNav = expectedElements['orgNav'];
+      expect(organizationNav).toBeInTheDocument();
     });
-  });
 
-  describe('it successfully constructs urls based on hosts and urls props', () => {
-    test('specific url overrides take precedence over hosts, when the prop is set', () => {
-      const accessManager = getByText('Access Manager').parentNode;
-      expect((accessManager as HTMLAnchorElement).href).toBe(
-        'https://cloud.mongodb.com/access-manager-test',
+    test('the project nav is rendered', () => {
+      const projectNav = expectedElements['projectNav'];
+      expect(projectNav).toBeInTheDocument();
+    });
+
+    test('current orgId is set based on data returned from fetch', () => {
+      const billing = expectedElements['billing'];
+      expect((billing as HTMLAnchorElement).href).toBe(
+        `https://cloud.mongodb.com/v2#/org/${dataFixtures?.currentOrganization?.orgId}/billing/overview`,
```

Comment:
We know this value should exist right? In that case we can do:

```suggestion
        `https://cloud.mongodb.com/v2#/org/${dataFixtures!.currentOrganization!.orgId}/billing/overview`,
```

The `!` is a TS feature that tells TS that I know this value exists.

Reply:
👍